libs/sdk-core/src/persist/swap.rs

```diff
@@ -329,24 +331,54 @@ impl SqliteStorage {
         self.select_single_swap("swaps.bitcoin_address = ?1", [address])
     }
 
-    pub(crate) fn list_swaps_with_status(
-        &self,
-        status: SwapStatus,
-    ) -> PersistResult<Vec<SwapInfo>> {
+    pub(crate) fn list_swaps(&self, req: ListSwapsRequest) -> PersistResult<Vec<SwapInfo>> {
         let con = self.get_connection()?;
-        let mut stmt = con.prepare(&self.select_swap_query("status = ?1", ""))?;
+        let mut where_clause = String::from("");
+        if let Some(status) = req.status {
+            if status.is_empty() {
+                return Ok(Vec::new());
+            }
+
+            append_and_clause(
+                &mut where_clause,
+                &format!(
+                    "status in ({})",
+                    status
+                        .into_iter()
+                        .map(|s| (s as u32).to_string())
+                        .collect::<Vec<_>>()
+                        .join(",")
+                ),
+            );
+        }
 
-        let vec: Vec<SwapInfo> = stmt
-            .query_map([status as u32], |row| self.sql_row_to_swap(row, ""))?
-            .map(|i| i.unwrap())
-            .collect();
+        if let Some(from_timestamp) = req.from_timestamp {
+            append_and_clause(
+                &mut where_clause,
+                &format!("created_at >= {}", from_timestamp),
+            );
+        }
 
-        Ok(vec)
-    }
+        if let Some(to_timestamp) = req.to_timestamp {
+            append_and_clause(&mut where_clause, &format!("created_at < {}", to_timestamp));
+        }
 
-    pub(crate) fn list_swaps(&self) -> PersistResult<Vec<SwapInfo>> {
-        let con = self.get_connection()?;
-        let mut stmt = con.prepare(&self.select_swap_query("true", ""))?;
+        if where_clause.is_empty() {
+            where_clause.push_str("true");
+        }
+
+        let mut query = self.select_swap_query(&where_clause, "");
+
+        match req.limit {
+            Some(limit) => query.push_str(&format!("LIMIT {}\n", limit)),
+            None => query.push_str("LIMIT -1\n"),
```

Comment:
NIT: do we have to put LMIT in this case?

Reply:
You can't have OFFSET only, so this is defense against the client putting only OFFSET